### ISSUE TITLE
MXE compilation problems

### ISSUE THREAD

**Issue by somot** [CleverRaven/Cataclysm-DDA#21248]
Hello. I'm trying to compile this game on Lubuntu 17.04 VirtualBox VM with Windows 7 Home Premium host. I'm using MXE and I'm getting the following error: 
```

src/iexamine.cpp: In function 'void iexamine::fswitch(player&, const tripoint&)':
src/iexamine.cpp:1298:6: error: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Werror=strict-overflow]
void iexamine::fswitch(player &p, const tripoint &examp)
```

The full log is here: http://www.wepaste.com/MXE%20compilation%20log/

Also, if I change the platform to i686 from x86_64, the following error appears: 
``
objwin/tiles/name.o: file not recognized: File format not recognized
``
And after performing `make clean` command, switching to i686 platform and compiling normally, the following error appears:

`/root/src/mxe/usr/bin/i686-w64-mingw32.static-ld: cannot find -lSDL2_ttf
`

Any help is very much welcome.

**Comment by narc0tiq** [comment]
Purely guessing, but:
- The first one might be an actual problem that our current set of compilers doesn't notice (or you might be compiling an old and broken commit).
- The second is probably a lack of `make clean` (it was trying to compile a 64-bit object file into a 32-bit binary), confirmed by it being fixed thereafter.
- The third suggests that your mxe only built `sdl2_ttf` for i386. Your mxe probably needs a recompile.

**Comment by somot** [comment]
Currently recompiling MXE, still waiting for success/failure.

**Comment by Graywolfe813** [comment]
This is totally a hack, but it works. Goes in your mxe/usr/[x86_64 || i686]-w64-mingw32.static/bin directory. This just disables the $SYSROOT check, something is wrong with it.

[freetype-config.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/1081969/freetype-config.zip)

**Comment by somot** [comment]
Do I need to recompile the MXE?

**Comment by Graywolfe813** [comment]
Just run make again, don't clean, and it'll build sdl2_ttf. I spaced it yesterday, but if you're going to use that for i686, you'll have to edit the paths in the file accordingly.

**Comment by narc0tiq** [comment]
Good(ish) news: I've updated the mxe on gorgon to the latest and now the build error you first encountered is part of the official builds -- so it should get fixed soon (in some fashion).

**Comment by kevingranade** [comment]
Fixed by #21286

**Comment by somot** [comment]
Danke Schön, Herren Kevin und Narc0tiq!
